resources/asciidoctor/spec/definition_hover_spec.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require 'definition_hover/extension'
+
+RSpec.describe ChangeAdmonition do
+  before(:each) do
+    Asciidoctor::Extensions.register DefinitionAdmonition
+    # These can't be rendered without docbook compat at this point.
+    Asciidoctor::Extensions.register DocbookCompat
+  end
+
+  after(:each) do
+    Asciidoctor::Extensions.unregister_all
+  end
+
+  include_context 'convert without logs'
+
+  shared_examples 'definition pop-up' do
+    context 'inline form' do
+      def expect_definition_popup(word, definition)
+        expect(converted).to include <<~HTML.strip
+          <span class="Definition Definition--definition" \
+          aria-describedby="Definition-defined">
+          <span class="Definition-word">#{word}</span>
+          <span class="Definition-defined">
+          #{definition}
+          </span>
+          </span>
+        HTML
+      end
+      context 'with text' do
+        let(:input) do
+          <<~ASCIIDOC
+            Words #{key}:[#{word},#{definition}] words.
+          ASCIIDOC
+        end
+        it "renders with Elastic's custom template" do
+          expect_definition_popup(
+            word.to_s, definition.to_s
+          )
+        end
+      end
+      context 'with a comma' do
+        let(:input) do
+          <<~ASCIIDOC
+            Words #{key}:[#{word},"#{definition} , comma"] words.
+          ASCIIDOC
+        end
+        it "renders with Elastic's custom template" do
+          expect_definition_popup(
+            word.to_s, "#{definition} , comma"
+          )
+        end
+      end
+      context 'without definition' do
+        let(:input) do
+          <<~ASCIIDOC
+            Words #{key}:[#{word},] words.
+          ASCIIDOC
+        end
+        it "renders with Elastic's custom template" do
+          expect_definition_popup(
+            word.to_s, ''
+          )
+        end
+      end
```

Comment:
This should really produce an error, but I'm not sure how to make that happen. A word and definition should be required

Reply:
Can we have the `Definition.process` method raise an error if either argument is a blank string? I don't know what built-in errors Asciidoctor defines, but it would be best to follow an existing pattern there.